elements/pf-back-to-top/pf-back-to-top.ts

```diff
@@ -0,0 +1,169 @@
+ *
+ * @cssprop {<length>} --pf-c-back-to-top--Right {@default `3rem``}
+ * @cssprop {<length>} --pf-c-back-to-top--Bottom {@default `1.5rem``}
+ * @cssprop --pf-c-back-to-top--c-button--BoxShadow {@default `0 0.75rem 0.75rem -0.5rem rgba(3, 3, 3, 0.18)`}
+ * @cssprop {<length>} --pf-c-button--FontSize {@default `0.75rem`}
+ * @cssprop {<length>|<percentage>}  --pf-c-button--BorderRadius {@default `30em`}
+ * @cssprop {<length>} --pf-c-button--PaddingTop {@default `0.25rem`}
+ * @cssprop {<length>} --pf-c-button--PaddingRight {@default `0.5rem`}
+ * @cssprop {<length>} --pf-c-button--PaddingBottom {@default `0.25rem`}
+ * @cssprop {<length>} --pf-c-button--PaddingLeft {@default `0.5rem`}
+ * @cssprop {<color>} --pf-c-button--m-primary--Color {@default `#fff`}
+ * @cssprop {<color>} --pf-c-button--m-primary--BackgroundColor {@default `#06c`}
+ * @cssprop {<length>} --pf-c-button__icon--m-end--MarginLeft {@default `0.25rem`}
+ *
+ */
+@customElement('pf-back-to-top')
+export class PfBackToTop extends LitElement {
+  static readonly styles = [styles];
+
+  /** Shorthand for the `icon` slot, the value is icon name */
+  @property({ reflect: true }) icon?: string;
+
+  /** Icon set for the `icon` property */
+  @property({ attribute: 'icon-set' }) iconSet?: string;
+
+  /** Flag to always show back to top button, defaults to false. */
+  @property({ reflect: true, type: Boolean, attribute: 'always-visible' }) alwaysVisible = false;
+
+  /** Element selector to spy on for scrolling. Not passing a selector defaults to spying on window scroll events */
+  @property({ reflect: true, attribute: 'scrollable-selector' }) scrollableSelector?: string;
+
+  /** Distance from the top of the scrollable element to trigger the visibility of the back to top button */
+  @property({ reflect: true, type: Number, attribute: 'scroll-distance' }) scrollDistance = 400;
+
+  /** Page fragment link to target element, must include hash ex: #top */
+  @property({ reflect: true }) href?: string;
+
+  #scrollSpy = false;
+
+  #visible = false;
+
+  #scrollElement?: Element | Window;
+
+  #logger = new Logger(this);
+
+  get #rootNode(): Document | ShadowRoot {
+    const root = this.getRootNode();
+    if (root instanceof Document || root instanceof ShadowRoot) {
+      return root;
+    } else {
+      return document;
+    }
+  }
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+    this.#setScrollListener();
+  }
+
+  override disconnectedCallback(): void {
+    super.disconnectedCallback?.();
+    this.#scrollElement?.removeEventListener('scroll', this.#toggleVisibility);
+  }
+
+  override updated(changed: PropertyValues<this>): void {
+    super.updated(changed);
+    if (changed.has('scrollableSelector')) {
+      if (changed.get('scrollableSelector') === undefined) {
+        this.#scrollElement?.removeEventListener('scroll', this.#toggleVisibility);
+      } else {
+        this.#setScrollListener();
+      }
+    }
+  }
```

Comment:
```suggestion
  override disconnectedCallback(): void {
    super.disconnectedCallback?.();
    this.#removeScrollListener();
  }

  override willUpdate(changed: PropertyValues<this>): void {
    if (changed.has('scrollableSelector')) {
      this.#addScrollListener();
    }
  }
```

This should be enough, but check the tests